src/core/qgsexpression.cpp

```diff
@@ -3991,6 +4024,7 @@ const QList<QgsExpression::Function*>& QgsExpression::Functions()
     << new StaticFunction( QStringLiteral( "make_point_m" ), 3, fcnMakePointM, QStringLiteral( "GeometryGroup" ) )
     << new StaticFunction( QStringLiteral( "make_line" ), -1, fcnMakeLine, QStringLiteral( "GeometryGroup" ) )
     << new StaticFunction( QStringLiteral( "make_polygon" ), -1, fcnMakePolygon, QStringLiteral( "GeometryGroup" ) )
+    << new StaticFunction( QStringLiteral( "make_triangle" ), -1, fcnMakeTriangle, QStringLiteral( "GeometryGroup" ) )
```

Comment:
Can you change this to use the new style which takes a ParameterList? That will correctly handle detection of invalid arguments.